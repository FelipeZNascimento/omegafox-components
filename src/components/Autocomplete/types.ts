export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
