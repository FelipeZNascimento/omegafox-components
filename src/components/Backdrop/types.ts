export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
