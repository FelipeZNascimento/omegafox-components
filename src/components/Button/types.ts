export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
