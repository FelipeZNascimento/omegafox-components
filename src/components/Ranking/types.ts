export type TRankingColumn = {
  id: number;
  align: 'left' | 'center' | 'right';
  flex: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
  renderingFunction: () => JSX.Element;
};

export type TRankingRowColumn = {
  id: number;
  renderingFunction: () => JSX.Element;
};

export type TRankingRow = TRankingRowColumn[];

export interface IRankingProps {
  backgroundImage?: string;
  isHeader: boolean;
  isLoading?: boolean;
  columns: TRankingColumn[];
  rows: TRankingRow[];
}
