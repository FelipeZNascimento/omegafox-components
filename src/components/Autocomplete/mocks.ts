import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
