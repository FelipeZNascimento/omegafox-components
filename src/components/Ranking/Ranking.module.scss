@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include shadowed(0.3);
  @include padded(all, l);
  @include flex-aligned(flex-start, flex-start);
  @include bg-rgba-colored($color-grey1, 0.5);

  background-size: contain;
  background-position: center;
  background-repeat: no-repeat;
  background-origin: content-box;
  border-radius: $border-radius;
  border-bottom-right-radius: 0;
  flex-direction: column;
  flex: 1;
  overflow-y: auto;

  .table,
  h1 {
    width: 100%;
  }
}

.table {
  &Header {
    @extend .table;
    @include flex-aligned(center, center);
    @include padded(vertical, s);
  }
}

.aligned {
  &Left {
    @include aligned(left);
  }

  &Center {
    @include aligned(center);
  }

  &Right {
    @include aligned(right);
  }
}

.flexed {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;

  & * {
    @extend .flexed;
  }

  &0 {
    @extend .flexed;
    @include flexed(0);
  }
  &1 {
    @extend .flexed;
    @include flexed(1);
  }
  &2 {
    @extend .flexed;
    @include flexed(2);
  }
  &3 {
    @extend .flexed;
    @include flexed(3);
  }
  &4 {
    @extend .flexed;
    @include flexed(4);
  }
  &5 {
    @extend .flexed;
    @include flexed(5);
  }
  &6 {
    @extend .flexed;
    @include flexed(6);
  }
  &7 {
    @extend .flexed;
    @include flexed(7);
  }
  &8 {
    @extend .flexed;
    @include flexed(8);
  }
  &9 {
    @extend .flexed;
    @include flexed(9);
  }
  &10 {
    @extend .flexed;
    @include flexed(10);
  }
}

.loadingContainer {
  width: 100%;
}
