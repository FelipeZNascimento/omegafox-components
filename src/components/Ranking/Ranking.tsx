import React from 'react';
import classNames from 'classnames';

import { IRankingProps, TRankingRow, TRankingRowColumn } from './types';
import styles from './Ranking.module.scss';

export const Ranking = ({
  backgroundImage,
  columns,
  isHeader,
  rows
}: IRankingProps) => {
  const numOfCols = columns.length;

  const renderHeader = (rows: TRankingRow) => {
    return (
      <div className={styles.tableHeader}>
        {rows.map((item: TRankingRowColumn, index: number) => {
          //Avoid breaking table by rendering more columns than it should
          if (index >= numOfCols) {
            return null;
          }

          const flexSize = columns[index].flex;
          const itemAlign = columns[index].align;

          const itemClass = classNames(styles[`flexed${flexSize}`], {
            [styles.alignedLeft]: itemAlign === 'left',
            [styles.alignedCenter]: itemAlign === 'center',
            [styles.alignedRight]: itemAlign === 'right'
          });

          return (
            <div key={item.id} className={itemClass}>
              {item.renderingFunction()}
            </div>
          );
        })}
      </div>
    );
  };

  return (
    <div
      className={styles.container}
      style={{
        backgroundImage: `url(${backgroundImage})`
      }}
    >
      {isHeader && renderHeader(columns)}
      {rows.map((row) => renderHeader(row))}
    </div>
  );
};
