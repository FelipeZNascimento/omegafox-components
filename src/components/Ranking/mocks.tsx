import { IRankingProps, TRankingRow } from './types';
const singleRow: TRankingRow = [
  { id: 0, renderingFunction: () => <div>01.</div> },
  {
    id: 1,
    renderingFunction: () => <div>Felipe Zanon Felipe Zanon Felipe Zanon</div>
  },
  { id: 2, renderingFunction: () => <div>999</div> }
];

export const tableConfig: IRankingProps = {
  isHeader: true,
  columns: [
    {
      id: 0,
      align: 'left',
      flex: 1,
      renderingFunction: () => <div>Pos</div>
    },
    {
      id: 1,
      align: 'left',
      flex: 4,
      renderingFunction: () => <div>Nome</div>
    },
    {
      id: 2,
      align: 'right',
      flex: 1,
      renderingFunction: () => <div>Pts.</div>
    }
  ],
  rows: [singleRow, singleRow, singleRow, singleRow, singleRow, singleRow]
};
