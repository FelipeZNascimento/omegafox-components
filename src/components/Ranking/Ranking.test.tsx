import React from 'react';
import { render } from '@testing-library/react';
import { Ranking } from './Ranking';
import { IRankingProps } from './types';
import { tableConfig } from './mocks';

describe('<Ranking />', () => {
  const renderComponent = ({ isHeader, columns, rows }: IRankingProps) =>
    render(<Ranking isHeader={isHeader} columns={columns} rows={rows} />);

  it('should render', () => {
    const { container } = renderComponent({
      isHeader: tableConfig.isHeader,
      columns: tableConfig.columns,
      rows: tableConfig.rows
    });

    expect(container).toMatchSnapshot();
  });
});
