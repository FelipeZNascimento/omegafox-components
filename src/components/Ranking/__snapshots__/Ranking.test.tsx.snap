// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ranking /> should render 1`] = `
<div>
  <div
    class="container"
    style="background-image: url(undefined);"
  >
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          Pos
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Nome
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          Pts.
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
    <div
      class="tableHeader"
    >
      <div
        class="flexed1 alignedLeft"
      >
        <div>
          01.
        </div>
      </div>
      <div
        class="flexed4 alignedLeft"
      >
        <div>
          Felipe Zanon Felipe Zanon Felipe Zanon
        </div>
      </div>
      <div
        class="flexed1 alignedRight"
      >
        <div>
          999
        </div>
      </div>
    </div>
  </div>
</div>
`;
